src/Moq/Mock.cs

```diff
@@ -259,42 +259,77 @@ private MockException TryVerifySetups(Func<Setup, MockException> verifySetup)
 			}
 		}
 
-		internal static void VerifyVoid(Mock mock, LambdaExpression expression, Times times, string failMessage)
+		internal static void Verify(Mock mock, LambdaExpression expression, Times times, string failMessage)
 		{
 			Guard.NotNull(times, nameof(times));
 
-			var (obj, method, args) = expression.GetCallInfo(mock);
-			ThrowIfVerifyExpressionInvolvesUnsupportedMember(expression, method);
-
-			var expectation = new InvocationShape(method, args);
-			VerifyCalls(GetTargetMock(obj, mock), expectation, expression, times, failMessage);
+			var parts = expression.Split();
+			Mock.VerifyRecursive(mock, expression, parts, times, failMessage, verifyLast: (part, targetMock) =>
+			{
+				var expectation = new InvocationShape(part.Method, part.Arguments);
```

Comment:
In the future, we might not need `InvocationShape` anymore. Or rather, we might want to fuse it with `LambdaExpressionPart` into something that represents a single-dot setup expression.